node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/LoadBalancers.java

```diff
@@ -0,0 +1,152 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.lb;
+
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.vespa.hosted.provision.Node;
+import com.yahoo.vespa.hosted.provision.NodeList;
+import com.yahoo.vespa.hosted.provision.NodeRepository;
+import com.yahoo.vespa.hosted.provision.node.NodeAcl;
+import com.yahoo.vespa.hosted.provision.persistence.CuratorDatabaseClient;
+
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * The load balancers of this node repo and the information (such asd ACL's) which need load balancer information.
+ *
+ * @author bratseth
+ * @author mpolden
+ */
+public class LoadBalancers {
+
+    private final CuratorDatabaseClient db;
+    private final NodeRepository nodeRepository;
+
+    public LoadBalancers(CuratorDatabaseClient db, NodeRepository nodeRepository) {
+        this.db = db;
+        this.nodeRepository = nodeRepository;
+    }
+
+    /** Returns a filterable list of all load balancers in this repository */
+    public LoadBalancerList list() {
+        return list((ignored) -> true);
+    }
+
+    /** Returns a filterable list of load balancers belonging to given application */
+    public LoadBalancerList list(ApplicationId application) {
+        return list((id) -> id.application().equals(application));
+    }
+
+    private LoadBalancerList list(Predicate<LoadBalancerId> predicate) {
+        return LoadBalancerList.copyOf(db.readLoadBalancers(predicate).values());
+    }
+
+    /**
+     * Returns the ACL for the node (trusted nodes, networks and ports)
+     */
+    private NodeAcl getNodeAcl(Node node, NodeList candidates) {
```

Comment:
This doesn't belong here. It creates a list of addresses and ports a node should trust, which is unrelated to load balancers (load balancer networks are just one of the things to trust).

It should be moved back, or into a separate class (`NodeAcls`?).

Reply:
I was afraid you'd say that because it is true.
But only logically true, since load balancers is the only *state* they depend on, and the tree of singleton objects below the node repo are all about state.
The more correct way is to put this below node and pass in the load balancers ... it's just more work. I'll do that then.

Reply:
> load balancers is the only state they depend on

They also depend on node state. `public List<NodeAcl> getNodeAcls(Node node, boolean children)` reads all nodes.